src/main/java/com/google/devtools/build/remote/client/LogParserUtils.java

```diff
@@ -258,9 +306,16 @@ private void printEntriesGroupedByAction(OutputStream outStream)
   /** Print log entries to standard output according to the command line arguments given. */
   public void printLog(PrintLogCommand options) throws IOException {
     try {
-      if (options.groupByAction) {
+      if (options.groupByAction && options.formatJson){
+        System.err.println("You can't use groupByAction with formatJson");
+      }
+      else if (options.groupByAction){
         printEntriesGroupedByAction(System.out);
-      } else {
+      }
+      else if (options.formatJson) {
+        printEntriesInJson(System.out);
```

Comment:
This feels a bit odd anyway. Why can't I have my array of objects grouped by action in json output as well? Nominally `printEntries` should have grouping *and* format parameterization, simplifying the two, and making the render of the object just a matter of providing the correct transformation to a correctly sorted collection.

Reply:
Thanks